--- omniauth_callbacks_controller.rb	2017-07-21 23:41:40.000000000 -0700
+++ omniauth_callbacks_controller-patched.rb	2017-07-23 23:24:41.000000000 -0700
@@ -5,7 +5,14 @@
   protect_from_forgery except: [:kerberos, :saml, :cas3]
 
   Gitlab.config.omniauth.providers.each do |provider|
-    define_method provider['name'] do
+    provider_name = provider['name']
+
+    # Allow for multiple OpenID Connect providers
+    if provider_name == 'openid_connect'
+        provider_name = provider['args']['name']
+    end
+
+    define_method provider_name do
       handle_omniauth
     end
   end
