diff --git a/person.rb.orig b/person.rb
index 4d6f8ac..1a2cb35 100644
--- a/person.rb.orig
+++ b/person.rb
@@ -8,6 +8,11 @@ module Gitlab
 
       attr_accessor :entry, :provider
 
+      def self.find_by_uuid(uuid, adapter)
+        uuid = Net::LDAP::Filter.escape(uuid)
+        adapter.user('entryUUID', uuid)
+      end
+
       def self.find_by_uid(uid, adapter)
         uid = Net::LDAP::Filter.escape(uid)
         adapter.user(adapter.config.uid, uid)
