--- ./lib/gitlab/o_auth/user.rb
+++ ./lib/gitlab/o_auth/user-patched.rb
@@ -119,6 +119,7 @@

       def find_ldap_person(auth_hash, adapter)
         Gitlab::LDAP::Person.find_by_uid(auth_hash.uid, adapter) ||
+          Gitlab::LDAP::Person.find_by_uuid(auth_hash.uid, adapter) ||
           Gitlab::LDAP::Person.find_by_email(auth_hash.uid, adapter) ||
           Gitlab::LDAP::Person.find_by_dn(auth_hash.uid, adapter)
       end
