--- ./lib/gitlab/auth/o_auth/user.rb
+++ ./lib/gitlab/auth/o_auth/user.rb.patched
@@ -122,6 +122,7 @@

         def find_ldap_person(auth_hash, adapter)
           Gitlab::Auth::LDAP::Person.find_by_uid(auth_hash.uid, adapter) ||
+            Gitlab::Auth::LDAP::Person.find_by_uuid(auth_hash.uid, adapter) ||
             Gitlab::Auth::LDAP::Person.find_by_email(auth_hash.uid, adapter) ||
             Gitlab::Auth::LDAP::Person.find_by_dn(auth_hash.uid, adapter)
         end
